src/Microsoft.Identity.Web/ClientInfo.cs

```diff
@@ -25,19 +25,11 @@ public static ClientInfo CreateFromJson(string clientInfo)
                 throw new ArgumentNullException(nameof(clientInfo), $"client info returned from the server is null");
             }
 
-            return DeserializeFromJson<ClientInfo>(Base64UrlHelpers.DecodeToBytes(clientInfo));
-        }
-
-        internal static T DeserializeFromJson<T>(byte[] jsonByteArray)
-        {
-            if (jsonByteArray == null || jsonByteArray.Length == 0)
-            {
-                return default;
-            }
+            var jsonByteArray = Base64UrlHelpers.DecodeToBytes(clientInfo);
 
-            using var stream = new MemoryStream(jsonByteArray);
-            using var reader = new StreamReader(stream, Encoding.UTF8);
-            return (T)JsonSerializer.Create().Deserialize(reader, typeof(T));
+            using MemoryStream stream = new MemoryStream(jsonByteArray);
+            using StreamReader reader = new StreamReader(stream, Encoding.UTF8);
+            return (ClientInfo)JsonSerializer.Create().Deserialize(reader, typeof(ClientInfo));
```

Comment:
JSON deserialization can always return NULL (since you don't know what you deserialize). But a nullable value type is not the same as a nullable reference type. So general purpose serialization helpers that just take a "T" won't work. I think @pmaytak 's suggestions (`where T : notnull`) might do it, I didn't research this enough.

Reply:
But from a code perspective, why have a general purpose deserialization method in a class that only deserializes ClientInfo? Makes the code hard to read...